### PR TITLE
Add property name to Dimension

### PR DESCRIPTION
**Pull request by mirca**
Implements #424 
cc @betatim

### CODE REVIEW

File: skopt/tests/test_space.py [hunk 1]
```diff
@@ -415,3 +415,25 @@ def test_real_distance():
     reals = Real(1, 10)
     for i in range(1, 10+1):
         assert_equal(reals.distance(4.1234, i), abs(4.1234 - i))
+
+
+@pytest.mark.parametrize("space, name",
+                         [(Real(1, 2, name="learning rate"), "learning rate"),
+                          (Integer(1, 100, name="no of trees"), "no of trees"),
+                          (Categorical(["red, blue"], name="colors"), "colors")])
+def test_dimension_name(space, name):
```
Comment (MechCoder) [inline skopt/tests/test_space.py]: I think `space` should be replaced with `dimension`?

Comment (mirca) [reply skopt/tests/test_space.py]: nice catch!

**Comment by betatim** [comment]
LGTM, not sure really why circleCI is unhappy. I clicked rebuild earlier but still broken. @MechCoder any ideas on this?

**Comment by mirca** [comment]
@betatim @MechCoder Looks like the CircleCI failure is unrelated to this PR(?)

**Comment by betatim** [comment]
Yes I also think it is unrelated. Hopefully #502 fixes this.

**Comment by mirca** [comment]
Rebased and everything is green, @betatim ;)

**Comment by betatim** [comment]
🎆

**Pull request merged**